website/project/views/node.py

```diff
@@ -207,9 +207,12 @@ def project_new_node(auth, node, **kwargs):
             'Your component was created successfully. You can keep working on the project page below, '
             'or go to the new <u><a href={component_url}>component</a></u>.'
         ).format(component_url=new_component.url)
-        if form.inherit_contributors.data and node.has_permission(user, ADMIN):
+        if form.inherit_contributors.data and (node.has_permission(user, ADMIN) or node.has_permission(user, WRITE)):
             for contributor in node.contributors:
                 new_component.add_contributor(contributor, permissions=node.get_permissions(contributor), auth=auth)
+                if contributor is user and not node.has_permission(contributor, 'admin'):
```

Comment:
I'm going to be super picky here primarily for readability. What if we re-do this for loop to set a permissions variable to the current permissions at the top, and if the contributor is the user, set the permissions to CREATOR_PERMISSIONS (which you'll have to import alongside WRITE and ADMIN).  Then add the contributor to the component with permissions=permissions. This will keep from doing multiple permissions sets and should be a little more clear to people coming in what is going on.